### ISSUE TITLE
Cop idea: Line Continuation in Expressions

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12338]
## Description

Avoid line continuation with `\` where not required. In practice, avoid using line continuations for anything but string concatenation.

https://rubystyle.guide/#no-trailing-backslash

## Behavior

```ruby


# bad (\ is not needed here)
result = 1 - \
         2

# bad (\ is required, but still ugly as hell)
result = 1 \
         - 2

# good
result = 1 -
         2

long_string = 'First part of the long string' \
              ' and second part of the long string'
```

**Comment by ydakuka** [comment]
For my project personally, I come across the following bad-smelling code in specs:

```ruby
subject :user do
  create(:user, advertiser_type, \
         advertiser: advertiser, \
         user_type: user_type)
end

subject :user do
  create(:user, advertiser_type,
         advertiser: advertiser, \
         user_type: user_type)
end
```

**Comment by koic** [comment]
It's implemented by `Style/RedundantLineContinuation` cop.

**Comment by ydakuka** [comment]
OK, but it does not detect [my examples](https://github.com/rubocop/rubocop/issues/12338#issuecomment-1793415141).
